common/src/com/intellij/plugins/haxe/util/HaxeCommonCompilerUtil.java

```diff
@@ -95,155 +100,229 @@ public static boolean compile(final CompilationContext context) {
       context.log(HaxeCommonBundle.message("module.0.is.excluded.from.compilation", context.getModuleName()));
       return true;
     }
+
+    if (!verifyProjectSettings(context)) {
+      return false;
+    }
+
+    final String workingPath = calculateWorkingPath(context);
+    if (! workingPath.isEmpty()) {
+      context.setErrorRoot(workingPath);
+    }
+    final File workingDirectory = new File(FileUtil.toSystemDependentName(workingPath));
+    if (!workingDirectory.exists()) {
+      if (!workingDirectory.mkdirs()) {
```

Comment:
Well, these if-conditions really could be combined, but I understand your embarrassing.  There is some "inexplicable prejudice"  about short-circuit evaluation. Developers prefer not to combine statements with action's result in one condition, because we do not fully believe if the action will not be executed if statement equals "false".

Reply:
Well, I'm not embarrassed.  And it's not about short-circuit evaluation, either.  I didn't combine them because line 113 is basically a test, and line 114 is some work that is to be performed, that just happens to return a boolean that we can test.  Certainly, I can combine them, but it doesn't read as well to me, -- or make better code, in my opinion --  and that's why I didn't.  

Perhaps I would be truer to my own philosophy here if I captured the boolean result and tested that with a separate if statement.